pilot/pkg/model/network.go

```diff
@@ -233,7 +234,7 @@ func (mgr *NetworkManager) resolveHostnameGateways(gatewaySet map[NetworkGateway
 	for host, addrs := range mgr.NameCache.Resolve(names) {
 		gwsForHost := hostnameGateways[host]
 		if len(addrs) == 0 {
```

Comment:
sorry I didn't notice t his on the first review - len(addrs)==0 is NOT an error IMO. We should distinguish from actual errors fetching an address and getting a succesful response with 0 IP addresses. IMO, in the later case, its not appropriate to retain the last know values

Caveat: I am not a DNS expert. Maybe we can check how Envoy STRICT_DNS cluster works and see if they agree with what I said

Reply:
Fixed
ref: https://github.com/envoyproxy/envoy/issues/20890